src/mscorlib/shared/System/String.cs

```diff
@@ -438,7 +438,11 @@ public unsafe char[] ToCharArray(int startIndex, int length)
         [NonVersionable]
         public static bool IsNullOrEmpty(string value)
         {
-            return (value == null || value.Length == 0);
+            // Using 0u >= (uint)value.Length rather than
+            // value.Length == 0 as it will elide the bounds check to
+            // the first char: value[0] if that is performed following the test
+            // for the same test cost.
+            return (value == null || 0u >= (uint)value.Length) ? true : false;
```

Comment:
The comment explains the use of 0u, but why use the (redundant) ternary operator instead of:
```C#
return (value == null || 0u >= (uint)value.Length);
```

Reply:
https://github.com/dotnet/coreclr/pull/17512#issuecomment-380427967

That does this change
```diff
        cmp      dword ptr [rsi+8], 0
-       sete     al
-       movzx    rax, al
-       test     eax, eax
```

Reply:
> That does this change

Thanks! I think we should highlight that in the comments too (to avoid accidentally reverting it in the future).

Reply:
@mikedn do you have an issue # for this?

Reply:
Added comment